javascript/javascript1/week4/Vioce.js

```diff
@@ -0,0 +1,122 @@
+
+// Vioce Assistant: 
+
+
+ const toDo=[]; 
+ let name; 
+ 
+ function getReply(command){
+
+let words = command.split(' '); 
+//console.log(words); 
+ 
+   if( command.startsWith('Hello my name is' ) || command.startsWith('Hi i am')){
+      
+         const lastWord= words.pop(''); 
+         const  name= lastWord ;
+          
+         if( name == 'Benjamin'){
+         
+         return'Nice to meet you'+' '+ name; 
+
+      }
+         if(name!=='Benjamin'){
+         return 'Nice to meet you' +' '+ name; 
+         
+      }
+        if (name == '' || name == undefined){
+       return 'What is your name?';
+
+    }
+      }
+   
+ //todo list: 
+
+    if (command.startsWith('add') && command.includes(' to my todo')){
+          const task= words.slice( 1 , words.length - 3).join(' ');
+          toDo.push( task) ; 
+          return task + 'is added to my todo';  
+         }
+         
+        else if ( command.startsWith('remove')&& command.includes('from my todo')){
+            const removeItem = toDo.slice( toDo[0] , 1); 
+            return removeItem + 'is  removed from my todo'; 
+
+        }
+      
+
+      // date: 
+
+      if( command.startsWith('what') && command.includes('today?')){
+
+         const today = new Date();
+         const  day = today.getDate();
+          const month = today.toLocaleString('en', {month:'long',}); 
+           const Year = today.getFullYear();
+           return 'today is '+ day +'. of ' + month +' '+  Year;
+       }
+      
+
+       //math : 
+       if (command.startsWith('what is ') && command.includes('+') || command.includes('-') || 
+           command.includes('*') ||command.includes('/')){
+               
+         if ( command.includes('+')){
+
+               return parseInt(words[2]) + parseInt(words[4]);
+         }
+         
+         if ( command.includes('-')){
+
+            return parseInt(words[2]) - parseInt(words[4]);
+      }
+      
+      if ( command.includes('*')){
+
+         return parseInt(words[2]) * parseInt(words[4]);
+   }
+   
+   if ( command.includes('/')){
+
+      return parseInt(words[2]) / parseInt(words[4]);
+}
+           }
+
+
+
+       // timer: 
+       if( command.startsWith('set')&& command.includes('a timer for')){
+
+       let minutes= words[words.length - 2];
+       let seconds= minutes * 60* 1000; 
+      
+       function setTimer(minutes){
+
+      console.log( 'timer is set for ' + minutes+' '+ 'minutes'); 
+      
+       }
+     setTimer(4); 
+       
+     function setTimeout(seconds){
+      if(seconds === 0 ){
+         console.log('Time is over'); 
+      }
+     }
+     setTimeout(0); 
+
```

Comment:
You do not need a word "function" in front of setTimeout function, because it is a global method, here you can read more about it : https://developer.mozilla.org/en-US/docs/Web/API/setTimeout

So solution for it will be :
` function setTimer(minutes){

   console.log( 'timer is set for ' + minutes+' '+ 'minutes'); 

   }
   setTimer(1); 

   function timerIsOverMessage(){
      console.log("Time is over")
   }
       
     setTimeout(timerIsOverMessage, seconds)`

And if your function "setTimer" is only for displaying information, that timer is set, I would recommend to use another name e.g. timerIsSetMessage, so you can avoid misunderstanding.